sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/models/CreateQueueOptions.java

```diff
@@ -0,0 +1,497 @@
+     * after which the message expires, starting from when the message is sent to Service Bus. This is the default value
+     * used when TimeToLive is not set on a message itself.
+     *
+     * @return the defaultMessageTimeToLive value.
+     */
+    public Duration getDefaultMessageTimeToLive() {
+        return this.defaultMessageTimeToLive;
+    }
+
+    /**
+     * Set the defaultMessageTimeToLive property: ISO 8601 default message timespan to live value. This is the duration
+     * after which the message expires, starting from when the message is sent to Service Bus. This is the default value
+     * used when TimeToLive is not set on a message itself.
+     *
+     * @param defaultMessageTimeToLive the defaultMessageTimeToLive value to set.
+     *
+     * @return the CreateQueueOptions object itself.
+     */
+    public CreateQueueOptions setDefaultMessageTimeToLive(Duration defaultMessageTimeToLive) {
+        this.defaultMessageTimeToLive = defaultMessageTimeToLive;
+        return this;
+    }
+
+    /**
+     * Get the deadLetteringOnMessageExpiration property: A value that indicates whether this queue has dead letter
+     * support when a message expires.
+     *
+     * @return the deadLetteringOnMessageExpiration value.
+     */
+    public boolean deadLetteringOnMessageExpiration() {
+        return this.deadLetteringOnMessageExpiration;
+    }
+
+    /**
+     * Set the deadLetteringOnMessageExpiration property: A value that indicates whether this queue has dead letter
+     * support when a message expires.
+     *
+     * @param deadLetteringOnMessageExpiration the deadLetteringOnMessageExpiration value to set.
+     *
+     * @return the CreateQueueOptions object itself.
+     */
+    public CreateQueueOptions setDeadLetteringOnMessageExpiration(boolean deadLetteringOnMessageExpiration) {
+        this.deadLetteringOnMessageExpiration = deadLetteringOnMessageExpiration;
+        return this;
+    }
+
+    /**
+     * Get the duplicateDetectionHistoryTimeWindow property: ISO 8601 timeSpan structure that defines the duration of
+     * the duplicate detection history. The default value is 10 minutes.
+     *
+     * @return the duplicateDetectionHistoryTimeWindow value.
+     */
+    public Duration getDuplicateDetectionHistoryTimeWindow() {
+        return this.duplicateDetectionHistoryTimeWindow;
+    }
+
+    /**
+     * Set the duplicateDetectionHistoryTimeWindow property: ISO 8601 timeSpan structure that defines the duration of
+     * the duplicate detection history. The default value is 10 minutes.
+     *
+     * @param duplicateDetectionHistoryTimeWindow the duplicateDetectionHistoryTimeWindow value to set.
+     *
+     * @return the CreateQueueOptions object itself.
+     */
+    public CreateQueueOptions setDuplicateDetectionHistoryTimeWindow(Duration duplicateDetectionHistoryTimeWindow) {
+        this.duplicateDetectionHistoryTimeWindow = duplicateDetectionHistoryTimeWindow;
```

Comment:
I see that generally, there is no Null check, is this by design ?

Reply:
Yes. I try to do as little client validation if the service can do it. No value is === "use the default service value". It's not an invalid value.